libs/ledger-live-common/src/families/filecoin/hw-signMessage.ts

```diff
@@ -1,10 +1,32 @@
 import Fil from "@zondax/ledger-filecoin";
 import { log } from "@ledgerhq/logs";
+import { CryptoCurrency } from "@ledgerhq/types-cryptoassets";
 
-import type { Resolver, Result } from "../../hw/signMessage/types";
+import type {
+  MessageData,
+  SignMessage,
+  Result,
+} from "../../hw/signMessage/types";
 import { getBufferFromString, getPath, isError } from "./utils";
 
-const resolver: Resolver = async (
+import type { DerivationMode } from "../../derivation";
+
+export const prepareMessageToSign = (
+  currency: CryptoCurrency,
+  path: string,
+  derivationMode: DerivationMode,
+  message: string
+): MessageData => {
+  return {
+    currency,
+    path,
+    derivationMode: derivationMode as DerivationMode,
+    message: Buffer.from(message, "hex").toString(),
+    rawMessage: "0x" + message,
+  };
+};
```

Comment:
A bit weird to have to define this, for families which potentially don't need it.
Any way to make it optional, maybe? Only call it if defined, basically?

Reply:
Ok, I thought it would be a good place, like the `signMessage` method.
Initially nothing was in the *family* code. Everything was in the common/signMessage file.
What is your preference or do you have another suggestion?

Reply:
side note: i know this works like this today so it is ok. but in future, we need to put this back into the AccountBridge because the ultimate goal is to drop all the magical "generated/*" files

Reply:
@gre , so the idea would be to add a method to the AccountBridge interface in LedgerJS?

Reply:
yeah, the overall goal should be to minimize a coin module to mostly just the bridges. bridge can also be renamed "API" if you want to understand this a bit. better. we have an api at the currency level and one at the account level (CurrencyBridge, AccountBridge), they respectively have either a currency or an account as a context.